emmalanguage/emma
# Code cleanup

In a (surely now) final stab towards preparing Emma for regular SNAPSHOT deployment, this PR does the following steps.

1. Removes code located under `eu.stratosphere` from all modules.
1. Removes GUI servlets associated with the old codebase.
1. Defines an `emma.onFlink` macro and defines a runner class and integration tests for the currently ported library methods.
1. Reorganizes `emma-language` into 
    1. `emma-language-library` - contains the `@emma.lib` function definitions;
    1. `emma-language-flink` - contains the `emma.onSpark` macro calls (Spark tests and runner);
    1. `emma-language-spark` - contains the `emma.onSpark` macro calls (Flink tests and runner);
1. Adds `addToClasspath()` to the `test.util` package and dynamically adds `/tmp/emma/codegen` to the classpath when necessary.
1. Simplifies POM in order to
    1. use `scalatest-maven-plugin`;
    1. run compiler tests with optional `compile` at the end of the pipelines using a `-Pcompile-spec-pipelines` profile;
    1. run integration tests in the `integration-test` phase.
1. Adapts the `.travis.yml` to run all tests in one pass using `mvn verify -Pcompile-spec-pipelines`.
